docs/examples/spectrum/spectral_factor.py

```diff
@@ -0,0 +1,140 @@
+# of the interval.
+
+# Create a location object
+lat, lon = metadata['latitude'], metadata['longitude']
+alt = altitude = metadata['altitude']
+tz = 'Etc/GMT+5'
+loc = location.Location(lat, lon, tz=tz, name='Greensboro, NC')
+
+# Calculate solar position parameters
+solpos = loc.get_solarposition(
+    meteo.index.shift(freq="-30min"),
+    pressure=meteo.pressure*100,  # convert from millibar to Pa
+    temperature=meteo.temp_air)
+solpos.index = meteo.index  # reset index to end of the hour
+
+amr = pvlib.atmosphere.get_relative_airmass(solpos.apparent_zenith).dropna()
+# relative airmass
+ama = amr*(meteo.pressure/1013.25)  # absolute airmass
+# %%
+# Now we calculate the clearsky index, :math:`k_c`, by comparing the TMY3 GHI
+# to the modelled clearsky GHI.
+
+cs = loc.get_clearsky(meteo.index)
+kc = pvlib.irradiance.clearsky_index(meteo.ghi, cs.ghi)
+# %%
+# :math:`W` is provided in the TMY3 file but in other cases can be calculated
+# from temperature and relative humidity
+# (see :py:func:`pvlib.atmosphere.gueymard94_pw`).
+
+w = meteo.precipitable_water
+
+# %%
+# Calculation of Spectral Mismatch
+# --------------------------------
+# Let's calculate the spectral mismatch factor using the three spectral factor
+# functions. First, we need to import some model coefficients for the SAPM
+# spectral factor function, which, unlike the other two functions, lacks
+# built-in coefficients.
+
+# Import some for a mc-Si module from the SAPM module database.
+module = pvlib.pvsystem.retrieve_sam('SandiaMod')['LG_LG290N1C_G3__2013_']
+# Calculate M using the three models for an mc-Si PV module.
+m_sapm = pvlib.spectrum.spectral_factor_sapm(ama, module)
+m_pvspec = pvlib.spectrum.spectral_factor_pvspec(ama, kc, 'multisi')
+m_fs = pvlib.spectrum.spectral_factor_firstsolar(w, ama, 'multisi')
+df_results = pd.concat([m_sapm, m_pvspec, m_fs], axis=1)
+df_results.columns = ['SAPM', 'PVSPEC', 'FS']
+# %%
+# Comparison Plots
+# ----------------
+# We can plot the results to visualise variability between the models. Note
+# that this is not an exact comparison since the exact same PV modules has
+# not been modelled in each case, only the same module technology.
+
+# Plot M
+fig1, (ax1, ax2) = plt.subplots(2, 1)
+df_results.plot(ax=ax1, legend=False)
+ax1.set_xlabel('Day')
+ax1.set_ylabel('Mismatch (-)')
+ax1.set_ylim(0.85, 1.15)
+
+# We can also zoom in one one day, for example August 2nd.
+df_results.loc['2001-08-02'].plot(ax=ax2)
+ax2.set_xlabel('Time')
+ax2.set_ylabel('Mismatch (-)')
+ax2.set_ylim(0.85, 1.15)
+
+plt.tight_layout()
+
+plt.show()
+
+# %%
+# References
+# ----------
+# .. [1] Daxini, Rajiv, and Wu, Yupeng (2023). "Review of methods to account
+#        for the solar spectral influence on photovoltaic device performance."
+#        Energy 286
+#        :doi:`10.1016/j.energy.2023.129461`
+# .. [2] Kasten, F. and Young, A.T., 1989. Revised optical air mass tables
```

Comment:
One of the references has the year in parentheses the other does not. Should be consistent.